modules/channel-web/src/views/lite/components/messages/Message.tsx

```diff
@@ -175,7 +175,23 @@ class Message extends Component<MessageProps> {
       return '* Cannot display message *'
     }
 
-    const type = this.props.type || (this.props.payload && this.props.payload.type)
+    let type = this.props.type || (this.props.payload && this.props.payload.type)
+
+    if (type === 'say_something') {
```

Comment:
instead of implementing the renderer logic in the render function, I believe that you should create a renderer for the say_something type, for example:

````javascript
render_say_something() {
    //Your logic
    return (this[`render_${text}`] || this.render_unsupported).call(this)
}
````